### PR TITLE
[Docs] Add common mistakes page

### PR DESCRIPTION
**Pull request by AugustinLF**
I observed that newcomers (based on this repo issues and difficulties that people in my team have) tend to encounter the same 4/5 issue. Perhaps a "common mistakes", troubleshooting or something like that could be helpful. So here is a simple proposition :)

I divided this section in two parts, firstly, the problems that people have by not understanding enough Flow's type system, and secondly, flow current limitations.

I struggle to understand how the website is built, so I just added the page, feel free to show me the right direction to make this change work!

Lastly, I saw that there is a "tips" folder in the docs, perhaps this page would better belong there, however, I don't think it's published, so I preferred to add it to the more common `lang` folder.

### CODE REVIEW

**Comment by mrkev** [comment]
IMO this is a great idea, sad it didn't get more eyes before. I'll try to import once the CI lets me

**Comment by AugustinLF** [comment]
Thanks @mrkev, I do think that a lot of PR for doc improvements flight under the radar, which is quite sad, the community being pretty motivated on that!

Regarding this PR, be wary, I created a page, without importing it in anywhere (or perhaps it's automatic?), so you might have to do it yourself, or point me to the right direction!

**Comment by mrkev** [comment]
I'll be completely honest, I'm well acquainted with how the website works. I think it's automatic though?

**Comment by AugustinLF** [comment]
Hey @mrkev, is there anything you'd like me to do to make this happen? :)

File: website/en/docs/lang/common-mistakes.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+layout: guide
+---
+
+Often, Flow will raise errors that you would consider wrong. Most of these issues are actually explained in the presentation of the [type system](https://flow.org/en/docs/lang/) and more specifically in the [soundness and completeness section](https://flow.org/en/docs/lang/types-and-expressions/#toc-soundness-and-completeness).
+Several of them are the consequence of careful tradeoffs made by the team, where you have to balance false positives and false negatives (see the above mentionned section).
+
+### Common user errors <a class="toc" id="toc-common-user-errors" href="#toc-common-user-errors"></a>
+
+#### Refinement invalidation
```
Comment (jbrown215) [inline website/en/docs/lang/common-mistakes.md]: I checked that `foo.bar` is not `null`, but Flow still thinks it is. Why does this happen?

File: website/en/docs/lang/common-mistakes.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+You have to keep in mind in JavaScript, you can modify data structures as you want, and Flow takes that in consideration. As of today, Flow cannot identify side-effects, so you end up with Flow considering that a `console.log()` can modify the content of an array (since nothing prevents you from modifying `console.log`). Therefore, whenever you refine types, some of your assumptions might not be valid anymore (this is especially common within closures). This is called [refinement invalidation](https://flow.org/en/docs/lang/refinements/#toc-refinement-invalidations).
```
Comment (jbrown215) [inline website/en/docs/lang/common-mistakes.md]: Can we add a basic refinement invalidation example here, too?

File: website/en/docs/lang/common-mistakes.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+#### Invalid subtyping
+
+Another difficulty people encounter is when they want to use a type different than the one declared that they believe compatible (as a parameter to a function for instance), and Flow complains, telling that the two types are not compatible.
+
+```js
+type WithError = {
```
Comment (jbrown215) [inline website/en/docs/lang/common-mistakes.md]: This example passes Flow :)

File: website/en/docs/lang/common-mistakes.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+type WithMaybeError = {
+    +resType: string,
+    +error: ?Object,
+};
+function handleAnswer(param: WithMaybeError) {}
+const res: WithError = {resType: 'withError', error: {code: 404}};
+handleAnswer(res); // Flow is happy!
+```
+
+### Errors due to Flow limitations <a class="toc" id="toc-flow-errors" href="#toc-flow-errors"></a>
```
Comment (jbrown215) [inline website/en/docs/lang/common-mistakes.md]: Can we change this section to "Why do my refinements disappear inside a closure?"

File: website/en/docs/lang/common-mistakes.md (hunk 3 above)
Comment (AugustinLF) [reply website/en/docs/lang/common-mistakes.md]: Good catch (I believe it used to fail)!

File: website/en/docs/lang/common-mistakes.md (hunk 4 above)
Comment (AugustinLF) [reply website/en/docs/lang/common-mistakes.md]: Hmmm, I'm not sure it's the same thing. Here, it's more about "you're right, but it doesn't work with flow implementation". I could add an example with closures (which are indeed another common problem).

Comment (jbrown215) [reply website/en/docs/lang/common-mistakes.md]: For sure, I just would prefer to get the heading down to a specific question if you can. There are lots of Flow limitations and this example doesn't cover all of them :)

**Comment by AugustinLF** [comment]
I pushed again! Sorry, git didn't get it when I renamed the file :'( I added examples about closures, and switched all the titles to questions. It should be better, and more beginner friendly!

**Comment by jbrown215** [comment]
Thanks for the PR! And sorry for the delay in my review!

**Comment by jbrown215** [comment]
I'm going to land this diff tomorrow! I expect to make minimal edits, but I like what you have here!

**Comment by jbrown215** [comment]
@AugustinLF: I ended up making it into it's own guide, so you can find it at flow.org/en/docs/ and click FAQ!

**Comment by AugustinLF** [comment]
Thanks a lot, pretty excited that we got this merged 😃

**Pull request closed without merging**